File: R_map_filter_reduce (menu)/py/menu.py
from my_utils import *
from menu_features import *
def main():
    nums = [] 
    opcao = -1
    while opcao != 0:
        show(menu())
        opcao = obter_inteiro_faixa(0,19,'Digite a opção desejada: ')

        if opcao == 1:
            nums = gerar_vetor()

        elif opcao == 2:
            nums = preencher_vetor_manual(nums)

        elif opcao == 3:
            nums = preencher_vetor_auto(nums)

        elif opcao == 4:
            mostrar_vetor(nums)

        elif opcao == 5:
            nums = elevar_vetor(nums)

        elif opcao == 6:
            contadora_de_numeros(nums)

        elif opcao == 7:
            mostrar_somatorio(nums)

        elif opcao == 8:
            exibir_media_e_mediana(nums)

        elif opcao == 9:
            maior_menor_numero(nums)

        elif opcao == 10:
            show(sorteio_positivo_negativo(nums))

        elif opcao == 11:
            gerar_n_grupos_tamanho_t(nums)

        elif opcao == 12:
            show(vetor_igual_ao_atual(nums))

        elif opcao == 13:
            show(top_n_maiores_numeros(nums))

        elif opcao == 14:
            show(top_n_menores_numeros(nums))

        elif opcao == 15:
            maiores_menores_que_a_media(nums)

        elif opcao == 16:
            show(chatinha(nums))

        elif opcao == 17:
            nums = ordenar_crescente_condicional(nums)

        elif opcao == 18:
            nums = ordenar_decrescente_condicional(nums)

        elif opcao == 19:
            nums = tirar_multplos_de_n_e_m(nums)

        elif opcao == 0:
            break
        input('\nEnter?')
        limpar_tela()
    limpar_tela()
    bye() 
        
def menu():
    return '''\n
 ===================================    MENU     ===================================================
| 01. Gerar vetor N posições
| 02. Preencher vetor manualmente item a item
| 03. Preencher vetor automaticamente
| 04. Mostrar vetor
| 05. Transformar vetor: elevar a potência N
| 06. Contar: Positivos, Negativos e Zeros
| 07. Somatório: De todos, Dos Negativos e dos Positivos
| 08. Exibir Média e Mediana: De Todos, Dos Positivos e Dos Negativos
| 09. Exibir Maior e Menor número
| 10. Sortear dois números: um positivo e um negativo
| 11. Gerar N grupos de T tamanhos. Não repetir valores
| 12. Pedir um novo vetor e verificar se está 100% presente nos números do sistema (e na mesma ordem)
| 13. Top N maiores números
| 14. Top N menores números
| 15. Listar valor médio, e listar números maiores que a Média e Menores que a Média
| 16. Soma da Média dos Números Positivos Múltiplos De dois COM
|   o Produto acumulado dos números negativos pares reduzidos à metade.
| 17. Ordenar os números em ordem crescente:
| 18. Ordenar os números em ordem decrescente
| 19. Eliminar números múltiplos de N e M (simultaneamente)
| 00. Sair
 ==================================================================================================='''
main() 

File: R_map_filter_reduce (menu)/java/Menu.java
import java.util.Scanner;
public class Menu{
    public static final int SAIR = 0;
    public static void main(String[] args) {
        
        String menu = StrUtils.gerarMenu("MENU", "Gerar vetor");
        int numeros[], opcao;
        do{
            opcao = MenuUtils.obterOpcao(menu,1);
            numeros = new int[3];
        }while(opcao != SAIR);
        switch(1){
        case 0:
            break;
        case 1:
            System.out.print("Digite o tamanho do vetor: ");
            Scanner input = new Scanner(System.in);
            int len = input.nextInt();
            numeros = new int[len];
            break;
        }
    }
    
}
/*1. Gerar vetor N posições, pedir valor padrão
Pedi N ao usuário e gera um vetor com todas as posições vazia (None ou undefined) ou com valor padrão se fornecido
python: vetor = [None] * N
javascript: vetor = Arrays(N)
Em seguida, use sua função mapear para preencher com o valor padrão, ou seja, transformar cada item no valor que o usuário pediu

2. Preencher vetor manualmente item a item
Pedi ao usuário sequencialmente cada dos valores (somente números)

3. Preencher vetor automaticamente
Peça ao usuário valores Mínimos e Máximos 

4. Mostrar vetor
Mostra os elementos do vetor

5. Transformar vetor: elevar a potência N
Substitui cada item por seu quadrado se expoente for 2, por exemplo

6. Contar: Positivos, Negativos e Zeros

7. Somatório: De todos, Dos Negativos e dos Positivos
Exibir as 3 informações sobre os valores

8. Exibir Média e Mediana: De Todos, Dos Positivos e Dos Negativos


9. Exibir Maior e Menor número


10. Sortear dois números: um positivo e um negativo


11. Gerar N grupos de T tamanhos. Não repetir valores


12. Pedir um novo vetor e verificar se está 100% presente nos números do sistema (e na mesma ordem)


13. Top N maiores números


14. Top N menores números


15. Listar valor médio, e listar números maiores que a Média e Menores que a Média


17. Somatório da Média dos Números Positivos múltiplos dois COM o Produto acumulado dos números negativos pares reduzidos à metade


18. Ordenar os números em ordem crescente: 
Pergunta se: todos, ou apenas pares, ou impares, ou positivos ou negativos, ou ainda apenas os múltiplos (positivos ou negativos) de N


19. Ordenar em ordem decrescente
Pergunta se: todos, ou apenas pares, ou impares, ou positivos ou negativos, ou ainda apenas os múltiplos (positivos ou negativos) de N



20. Eliminar números múltiplos de N e M (simultaneamente)


0. Sair (mensagem automática de tchau) */